### PR TITLE
Fix incorrect doc for stdio_put_string

### LINKED ISSUE #2818: Docs: unclear description for `stdio_put_string`
The function doc states "This method returns as soon as input is available, but more characters may be returned up to the end of the buffer", but that doesn't make sense as this is an output-only function, I believe it might have been mistakenly pasted from the documentation of an input procedure such as `stdio_get_until`.

The function implementation looks like it won't return early if buffers are full, so it would be good to make that clear, since there is a returned `len`.

### PR DESCRIPTION
**Pull request by anoshyn**
## Summary

The doxygen for `stdio_put_string` in `src/rp2_common/pico_stdio/include/pico/stdio.h` reads *"This method returns as soon as input is available, but more characters may be returned up to the end of the buffer."* That text was copy-pasted from an input-reading helper (e.g. `stdio_get_until`); `stdio_put_string` is output-only and never returns early on its own. The wording is actively misleading because the function has an `int` return value, so a reader could plausibly assume some short-write behavior.

Rewrite the description to match the implementation in `stdio.c:92-114`:

- writes the buffer to every registered driver in turn (no early return);
- documents the `len == -1` → `strlen` convention;
- documents `cr_translation` behavior (applies to any appended newline too);
- documents the `0`-return edge case when `PICO_STDIO_IGNORE_NESTED_STDOUT` is set and a concurrent stdout write is in progress.

No behavior change; doxygen only.

Fixes #2818.

## Test plan

- [x] `git diff` — only `src/rp2_common/pico_stdio/include/pico/stdio.h` is touched; only the doxygen block for `stdio_put_string`.
- [ ] CI builds pass (header still compiles cleanly).
- [x] Doxygen renders without warnings for `stdio_put_string`.

### CODE REVIEW

**Comment by anoshyn** [comment]
@kilograham @lurch when you have a moment, would you mind taking a look? This is a doxygen-only change to `stdio_put_string` that fixes the description copy-paste flagged in #2818. Happy to adjust wording if you'd prefer different phrasing. Thanks!

**Comment by lurch** [comment]
Hi @anoshyn , as requested in other PRs, if you could update **all** of your pico-sdk PRs to remove the "Co-Authored-By: Claude" lines, that'd be helpful :+1:

File: src/rp2_common/pico_stdio/include/pico/stdio.h [hunk 1]
```diff
@@ -175,14 +175,20 @@ int stdio_get_until(char *buf, int len, absolute_time_t until);
 /*! \brief Prints a buffer to stdout with optional newline and carriage return insertion
  * \ingroup pico_stdio
  *
- * This method returns as soon as input is available, but more characters may
- * be returned up to the end of the buffer.
+ * Writes the first \p len characters of \p s to every registered stdio driver in
+ * turn. If \p len is `-1`, the length is taken from `strlen(s)`. When \p newline
+ * is true, a single line-feed character is appended after the buffer. When
+ * \p cr_translation is true, each line-feed written (including any appended
+ * newline) is converted to a carriage-return + line-feed pair. This call does
```
Comment (lurch) [inline src/rp2_common/pico_stdio/include/pico/stdio.h]: Feels like much of this is just duplicating information that's already covered in the `\param` descriptions below?

**Comment by anoshyn** [comment]
Force-pushed: dropped the `Co-Authored-By` trailer, and trimmed the prose body so it no longer restates the `\param` lines - now just a brief plus two short sentences covering what isn't in the params (full-buffer write, may block under back-pressure). New HEAD: c2a8ef1e.

**Review by kilograham — DISMISSED** [review]

File: src/rp2_common/pico_stdio/include/pico/stdio.h [hunk 2]
```diff
@@ -172,17 +172,19 @@ void stdio_set_chars_available_callback(void (*fn)(void*), void *param);
  */
 int stdio_get_until(char *buf, int len, absolute_time_t until);
 
-/*! \brief Prints a buffer to stdout with optional newline and carriage return insertion
+/*! \brief Prints a buffer to stdout, optionally appending a newline
  * \ingroup pico_stdio
  *
- * This method returns as soon as input is available, but more characters may
- * be returned up to the end of the buffer.
+ * Writes \p s in full to every registered stdio driver. The call returns once
```
Comment (lurch) [inline src/rp2_common/pico_stdio/include/pico/stdio.h]: I think I'd be tempted to do:
```suggestion
 * Writes \p s to every registered stdio driver. The call returns once
```
as you could give a `len` shorter than `s` if you didn't want to write all of it. (But this is optional, given that Graham has already approved this PR.)

Comment (anoshyn) [reply src/rp2_common/pico_stdio/include/pico/stdio.h]: Good catch — applied in 3dc5e04. "in full" did contradict the `\param len` line right below it.

**Review by kilograham — APPROVED** [review]
